NitroxClient/MonoBehaviours/DiscordRP/DiscordRPController.cs

```diff
@@ -0,0 +1,153 @@
+﻿using NitroxClient.MonoBehaviours.Gui.MainMenu;
+using NitroxModel.Core;
+using NitroxModel.Helper;
+using NitroxModel.Logger;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace NitroxClient.MonoBehaviours.DiscordRP
+{
+    public class DiscordRPController : MonoBehaviour
+    {
+        public DiscordRpc.RichPresence Presence = new DiscordRpc.RichPresence();
+        public bool ShowingWindow;
+
+        private const string APPLICATION_ID = "405122994348752896";
+        //private const string STEAM_ID = "264710"; Can cause bugs with Discord trying to start clean Subnautica without Nitrox
+        private DiscordRpc.EventHandlers handlers;
+
+        private static DiscordRPController main;
+        public static DiscordRPController Main
+        {
+            get
+            {
+                if (main == null)
+                {
+                    main = new GameObject("DiscordController").AddComponent<DiscordRPController>();
+                }
+                return main;
+            }
+        }
+
+        public void ReadyCallback(ref DiscordRpc.DiscordUser connectedUser) => Log.Info("[Discord] Ready");
+
+        public void DisconnectedCallback(int errorCode, string message) => Log.Info($"[Discord] Disconnected: {errorCode} <=> {message}");
+
+        public void ErrorCallback(int errorCode, string message) => Log.Error($"[Discord] Error: {errorCode} <=> {message}");
+
+        public void SpectateCallback(string secret) => Log.Info($"[Discord] Spectate: {secret}");
```

Comment:
What does the secret represent?  Should it be logged?

Reply:
For now it is the ip of the sever. So probably better to make it Log.Debug()?